website/server/controllers/api-v3/groups.js

```diff
@@ -471,29 +471,34 @@ api.joinGroup = {
 
     let isUserInvited = false;
 
-    if (group.type === 'party' && group._id === user.invitations.party.id) {
-      inviter = user.invitations.party.inviter;
-      user.invitations.party = {}; // Clear invite
-      user.markModified('invitations.party');
-
-      // invite new user to pending quest
-      if (group.quest.key && !group.quest.active) {
-        user.party.quest.RSVPNeeded = true;
-        user.party.quest.key = group.quest.key;
-        group.quest.members[user._id] = null;
-        group.markModified('quest.members');
-      }
+    // Check if was invited to party
+    if (group.type === 'party') {
+      if (_.find(user.invitations.parties, {id: group._id})) {
+        inviter = user.invitations.party.inviter;
+
+        // Clear all invitations of new user
+        user.invitations.parties = [];
+        user.invitations.party = {};
+
+        // invite new user to pending quest
+        if (group.quest.key && !group.quest.active) {
+          user.party.quest.RSVPNeeded = true;
+          user.party.quest.key = group.quest.key;
+          group.quest.members[user._id] = null;
+          group.markModified('quest.members');
+        }
 
-      // If user was in a different party (when partying solo you can be invited to a new party)
-      // make him leave that party before doing anything
-      if (user.party._id) {
-        let userPreviousParty = await Group.getGroup({user, groupId: user.party._id});
-        if (userPreviousParty) await userPreviousParty.leave(user);
-      }
+        // If user was in a different party (when partying solo you can be invited to a new party)
+        // make him leave that party before doing anything
```

Comment:
Ah yes, thanks @khipkin.  @Yutsuten We use [they / them as a singular pronoun](https://en.wikipedia.org/wiki/Singular_they) throughout Habitica to cover people of all genders.